File: src/test/java/com/nickmcdowall/lsd/interceptor/rest/StubHttpRequest.java
package com.nickmcdowall.lsd.interceptor.rest;

import lombok.Builder;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpHeaders;
import org.springframework.http.HttpRequest;

import java.net.URI;

@Builder
@RequiredArgsConstructor
public class StubHttpRequest implements HttpRequest {

    private final String methodValue;
    private final URI uri;
    private final HttpHeaders httpHeaders;

    @Override
    public HttpHeaders getHeaders() {
        return httpHeaders;
    }

    @Override
    public String getMethodValue() {
        return methodValue;
    }

    @Override
    public URI getURI() {
        return uri;
    }
}

File: lsd-interceptors/src/test/kotlin/io/lsdconsulting/interceptors/http/StubHttpRequest.kt
package io.lsdconsulting.interceptors.http

import org.springframework.http.HttpHeaders
import org.springframework.http.HttpMethod
import org.springframework.http.HttpRequest
import java.net.URI

data class StubHttpRequest(
    private val methodValue: String,
    private val uri: URI,
    private val httpHeaders: HttpHeaders
) : HttpRequest {
    override fun getHeaders(): HttpHeaders = httpHeaders
    override fun getMethod(): HttpMethod = HttpMethod.valueOf(methodValue)
    override fun getURI(): URI = uri
}
